### ISSUE TITLE
Roadmap of MMPose 1.x

### ISSUE THREAD

**Issue by Tau-J** [open-mmlab/mmpose#2258]
## Development Plan 2023
We are excited to announce the release of MMPose 1.0.0 as a part of the OpenMMLab 2.0 project!

MMPose v1.0.0 is a major update, including many API and config file changes, and most of the code (and config files) in MMPose 1.0 will **not be compatible** with 0.x version.

Here are the version correspondences between `mmdet`, `mmcv` and `mmpose`:
- MMDetection 2.x <=> MMPose 0.x <=> MMCV 1.x
- MMDetection 3.x <=> MMPose 1.x <=> MMCV 2.x

### Migration

Currently, a part of the algorithms have been migrated to v1.0.0, and the remaining algorithms will be completed in subsequent versions. We will show the migration progress in the following list.

| Algorithm                         |   Status    |
| :-------------------------------- | :---------: |
| MTUT (CVPR 2019)                  |             |
| MSPN (ArXiv 2019)                 |    done     |
| InterNet (ECCV 2020)              |             |
| DEKR (CVPR 2021)                  |    done     |
| HigherHRNet (CVPR 2020)           |             |
| DeepPose (CVPR 2014)              |    done     |
| RLE (ICCV 2021)                   |    done     |
| SoftWingloss (TIP 2021)           |             |
| VideoPose3D (CVPR 2019)           | in progress |
| Hourglass (ECCV 2016)             |    done     |
| LiteHRNet (CVPR 2021)             |    done     |
| AdaptiveWingloss (ICCV 2019)      |    done     |
| SimpleBaseline2D (ECCV 2018)      |    done     |
| PoseWarper (NeurIPS 2019)         |             |
| SimpleBaseline3D (ICCV 2017)      | in progress |
| HMR (CVPR 2018)                   |             |
| UDP (CVPR 2020)                   |    done     |
| VIPNAS (CVPR 2021)                |    done     |
| Wingloss (CVPR 2018)              |             |
| DarkPose (CVPR 2020)              |    done     |
| Associative Embedding (NIPS 2017) | in progress |
| VoxelPose (ECCV 2020)             |             |
| RSN (ECCV 2020)                   |    done     |
| CID (CVPR 2022)                   |    done     |
| CPM (CVPR 2016)                   |    done     |
| HRNet (CVPR 2019)                 |    done     |
| HRNetv2 (TPAMI 2019)              |    done     |
| SCNet (CVPR 2020)                 |    done     |

If your algorithm has not been migrated, you can continue to use the [0.x branch](https://github.com/open-mmlab/mmpose/tree/0.x) and [old documentation](https://mmpose.readthedocs.io/en/0.x/).

### New Algorithms
[x] ViTPose in https://github.com/open-mmlab/mmpose/pull/1876
[x] RTMPose [\[projects\]](https://github.com/open-mmlab/mmpose/tree/main/projects/rtmpose) in https://github.com/open-mmlab/mmpose/pull/1971
[x] YOLO-Pose [\[projects\]](https://github.com/open-mmlab/mmpose/tree/main/projects/yolox-pose) in https://github.com/open-mmlab/mmpose/pull/2020
[ ] PETR (in progress)

### New Datasets
[ ] DeepFasion2 in https://github.com/open-mmlab/mmpose/pull/2201 (in progress)
[ ] Human-Art [\[github\]](https://github.com/IDEA-Research/HumanArt) (in progress)

**Comment by Ben-Louis** [comment]
How about supporting [UniFormer](https://github.com/Sense-X/UniFormer/tree/main/pose_estimation)?

**Comment by tojimahammatov** [comment]
Hi, is there any news on migrating Associative Embedding algorithm? It would be very great to see it in 1.x version of mmpose. Very important algorithm in most of the models. Looking forward to having it. Or can you hint some ways to implement? I just need it for one of my ongoing researches. Thank you.

**Comment by tojimahammatov** [comment]
@Tau-J hello, sorry for the ping, is there any update on associative embedding? Could you please answer my questions above? Thank you.

**Comment by Tau-J** [comment]
> @Tau-J hello, sorry for the ping, is there any update on associative embedding? Could you please answer my questions above? I was working on an important task that is related to launch some services to a production. Thank you.

Sorry for the late reply. The algorithm of 'Associate Embedding' is still in the process of migration. As the community contributors responsible for this task is not a full-time developer, we cannot guarantee a specific completion timeline. If you are also willing to contribute to the migration, we would greatly appreciate it. You can follow our [1.x documentation](https://mmpose.readthedocs.io/en/dev-1.x/guide_to_framework.html) to understand the new version architecture and proceed with migrating the algorithm. If you meet any issue during migration, please feel free to contact us.

**Comment by tojimahammatov** [comment]
> > @Tau-J hello, sorry for the ping, is there any update on associative embedding? Could you please answer my questions above? I was working on an important task that is related to launch some services to a production. Thank you.
> 
> Sorry for the late reply. The algorithm of 'Associate Embedding' is still in the process of migration. As the community contributors responsible for this task is not a full-time developer, we cannot guarantee a specific completion timeline. If you are also willing to contribute to the migration, we would greatly appreciate it. You can follow our [1.x documentation](https://mmpose.readthedocs.io/en/dev-1.x/guide_to_framework.html) to understand the new version architecture and proceed with migrating the algorithm. If you meet any issue during migration, please feel free to contact us.

Never mind, thanks for the reply. I got it, I will try to migrate and contribute then. I do contact if I need some help or issue. Thank you.

**Comment by Ben-Louis** [comment]
> When will motionbert be merged into the main branch? Currently, I only see it in the dev branch.

Yes. The `dev-1.x` branch will be merged into the `main` branch recently

**Comment by thehiddenwaffle** [comment]
It would be really wonderful for RTM's wholebody applications if pose lifting was available using something like one of the models referenced in [H3WB](https://github.com/wholebody3d/wholebody3d/blob/main/benchmark.md), though I'm inexperienced and don't know if this is viable.

**Comment by gdtiti** [comment]
请问支持unity插件的事情有开始实施吗？

**Comment by tojimahammatov** [comment]
@Tau-J @Ben-Louis hi guys, sorry for the ping. We've been working with pose estimation task for some time. Until now, we could achieve some feasible results by successfully adapting pose estimation models to our domain space. However, it seems that there are still things to improve. A few issues like track fragmentation (wrong track is found for short time), and detecting tracks with black dress/shirts etc. can be observed from time to time. 
Can I ask you to make a contact or a call for at least consulting for us? Our company representatives and tech team will be very glad to listen to your invaluable feedbacks and ideas. I know that you also work for full-time and you must be too busy. But, I believe that it would cause a great progress for our development. My email is tojimahammatov@gmail.com. We look forward to hearing from you soon. Thanks.

**Comment by hariouat** [comment]
Hello, please how to test with new mmpose 1.x? how to use de dump pkl file ? how to convert pkl file to json file? 
Thanks in advance

**Comment by thehiddenwaffle** [comment]
> 请问可以实现根据RGB图对应的深度图还原出3D人体姿态的准确深度信息吗？
> 
> Excuse me, can the accurate depth information of 3D human body posture be restored according to the depth map corresponding to the RGB map?

RTMPose3D does this, however the depth is still relative to the cropped detection so if you need world coordinates then you will need some sort of scaling factor whether that be a known bone length, or a known Z value with a known camera calibration to reproject the points, etc.
